engine/kv_engine.hpp

```diff
@@ -92,26 +108,40 @@ class KVEngine : public Engine {
     return xPush(collection_name, value, QueueOpPosition::Right);
   }
 
-  void ReleaseWriteThread() override { write_thread.Release(); }
+  void ReleaseAccessThread() override { access_thread.Release(); }
 
   const std::vector<std::shared_ptr<Skiplist>> &GetSkiplists() {
     return skiplists_;
   };
 
 private:
+  friend OldRecordsCleaner;
+
+  KVEngine(const Configs &configs)
```

Comment:
Confused parameters compared with `Init`. Consider only pass `configs.max_access_thread`?

Reply:
it's a private constructor so I think it's ok